work16/src/main.js

```diff
@@ -0,0 +1,298 @@
+  const img = document.createElement("img");
+
+  div.classList = "loading_wrap";
+  div.id = "js_loading_wrap"
+  img.src = "https://i.postimg.cc/k2RLrq7B/loading-circle.gif";
+  img.alt = "ローディング画像"
+
+  div.appendChild(img);
+  document.body.prepend(div);
+}
+
+const renderNewsTab = (newsDataArray) => {
+  const tabList = document.getElementById("js_tabList");
+  const fragment = document.createDocumentFragment();
+
+  newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    let isInitialDisplay = newsCategoryObj.isInitialDisplay;
+
+    if (isInitialDisplay === "true") {
+      li.setAttribute("aria-selected", true);
+    } else {
+      li.setAttribute("aria-selected", false);
+    }
+    li.setAttribute("roll", "tab");
+    li.id = `${"js_tabTopics" + ++newsCategoryIndex}`;
+    li.classList = "tabTopics";
+
+    a.textContent = newsCategoryObj.field
+    a.href = "#"
+
+    li.appendChild(a);
+    fragment.appendChild(li);
+  });
+  tabList.appendChild(fragment);
+
+  clickEventChangeTabAttribute();
+}
+
+const renderNewsContent = (newsDataArray) => {
+  newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+    const tabTopics = document.querySelectorAll(".tabTopics");
+    const newsContentsData = newsCategoryObj.contents;
+    const newsCategoryImg = newsCategoryObj.img;
+    let isInitialDisplay = newsCategoryObj.isInitialDisplay;
+
+    if (isInitialDisplay === "true") {
+      const div = document.createElement("div");
+      const newsSection = createSection(tabTopics, newsCategoryIndex);
+
+      div.classList = "tabpanelTopics_wrap";
+
+      document.getElementById("js_tabList").after(newsSection);
+      div.appendChild(createTopicImg(newsCategoryImg));
+      div.prepend(renderArticle(newsContentsData, newsCategoryIndex));
+
+      newsSection.appendChild(div);
+    }
+
+    tabTopics[newsCategoryIndex].addEventListener("click", (e) => {
+      removeCommentIcon();
+      removeNewIcon();
+
+      newsContentsData.forEach((newsArticleData, categoryNewsArticleDataIndex) => {
+        const commentArray = newsArticleData.comments;
+        const liElements = document.querySelectorAll(".tabpanelTopics_wrap li");
+        const liElement = liElements[categoryNewsArticleDataIndex];
+
+        changeCategoryConetent(newsCategoryIndex, e.currentTarget);
+        changeCategoryTitle(newsArticleData, categoryNewsArticleDataIndex);
+        changeCategoryImg(newsCategoryImg);
+
+        if (commentArray.length !== 0) {
+          liElement.append(createCommentIcon(commentArray));
+        }
+
+        if (isLatestArticles(newsArticleData) === true) {
+          const h1Elements = document.querySelectorAll(".tabpanelTopics_wrap h1");
+          const h1Element = h1Elements[categoryNewsArticleDataIndex];
+          h1Element.insertAdjacentElement("afterend", createNewIcon());
+        }
+      })
+    })
+  })
```

Comment:
IMO
I feel it is not appropriate to have the tabTopics event definition within renderNewContent.
It would be better to define them together in clickEventChangeTabAttribute.

Reply:
Indeed, it was a mistake to have a tab switching event function in the renderNewsContent function.

I also thought that putting the tabTopics event definition inside the clickEventChangeTabAttribute function would make the function name strange, so I changed it to the clickedTabEvent function.

commit: [e58e83b](https://github.com/yasu-pro/js-handson/pull/18/commits/e58e83b07136394ca81dda8e3c990b38a2a5e36a)